src/steamship/experimental/transports/telegram.py

```diff
@@ -0,0 +1,111 @@
+import logging
+import tempfile
+from typing import Optional
+
+import requests
+
+from steamship import SteamshipError
+from steamship.experimental.transports.chat import ChatMessage
+from steamship.experimental.transports.transport import Transport
+
+API_BASE = "https://api.telegram.org/bot"
+
+
+class TelegramTransport(Transport):
+    """Experimental base class to encapsulate a Telegram communication channel."""
+
+    api_root: str
+
+    def __init__(self, bot_token: str):
+        self.api_root = f"{API_BASE}{bot_token}"
+
+    def _instance_init(self, *args, **kwargs):
+        if "webhook_url" not in kwargs:
+            raise SteamshipError(
+                message="Please provide the `webhook_url` method in the TelegramTransport instance_init kwargs"
+            )
+
+        webhook_url = kwargs["webhook_url"]
+        logging.info(f"Setting Telegram webhook URL: {webhook_url}")
+        logging.info(f"Post is to {self.api_root}/setWebhook")
+
+        response = requests.get(
+            f"{self.api_root}/setWebhook",
+            params={
+                "url": webhook_url,
+                "allowed_updates": ["message"],
+                "drop_pending_updates": True,
+            },
+        )
+
+        if not response.ok:
+            raise SteamshipError(
+                f"Could not set webhook for bot. Webhook URL was {webhook_url}. Telegram response message: {response.text}"
+            )
+
+    def _instance_deinit(self, *args, **kwargs):
+        """Unsubscribe from Telegram updates."""
+        requests.get(f"{self.api_root}/deleteWebhook")
+
+    def _send(self, blocks: [ChatMessage]):
+        """Send a response to the Telegram chat."""
+        for block in blocks:
+            chat_id = block.get_chat_id()
+            if block.is_text() or block.text:
+                params = {"chat_id": int(chat_id), "text": block.text}
+                requests.get(f"{self.api_root}/sendMessage", params=params)
+            elif block.is_image():
+                _bytes = block.raw()
```

Comment:
note: there is a size limit for Telegram uploads. we should probably add handling for any sort of common util for not trying to send anything too large.

Reply:
Ah, good to know. I'll fiddle with it and push up an update